[Swift in WebKit] Work towards modularizing WebCore private headers (Part 2)

rr-codes opened the pull request:
#### 9f2ca0320ff28a530dfdab33d763a5b3e8d109bc
<pre>
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 2)
<a href="https://bugs.webkit.org/show_bug.cgi?id=297503">https://bugs.webkit.org/show_bug.cgi?id=297503</a>
<a href="https://rdar.apple.com/158487517">rdar://158487517</a>

Reviewed by Abrar Rahman Protyasha.

- Fix some more quoted imports to angle bracket imports
- Convert some more private headers to project that don&apos;t need to be private, and vice versa
- Add some missing headers to the Xcode project
- Add a few missing imports
- Let Xcode fix some erroneous file character encoding (macOS Western -&gt; UTF-8)

* Source/WebCore/PAL/pal/text/TextEncoding.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AXTextMarker.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h:
* Source/WebCore/animation/AnimationFrameRatePreset.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/editing/cocoa/AlternativeTextContextController.h:
* Source/WebCore/editing/cocoa/AlternativeTextUIController.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/layout/floats/PlacedFloats.h:
* Source/WebCore/layout/layouttree/LayoutBoxGeometry.h:
* Source/WebCore/page/PerformanceEventTiming.h:
* Source/WebCore/page/ProcessSyncData.in:
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/platform/AudioEncoder.h:
* Source/WebCore/platform/graphics/AV1Utilities.h:
* Source/WebCore/platform/graphics/MediaPlayerEnums.h:

Canonical link: <a href="https://commits.webkit.org/298877@main">https://commits.webkit.org/298877@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/336130952f8fc6927ec6a0ff4a722a9a4e2890ba

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116930 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36595 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27169 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123015 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/68946 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a40133e8-6ade-419e-8cef-71230ad2b5b5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118819 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37292 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45195 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88809 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/43540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f4946d75-39cd-44dd-a55c-a70da28eaf5d) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119879 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29741 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104899 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69268 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28808 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23006 "Found 1 new test failure: inspector/worker/dom-debugger-event-after-terminate-crash.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66671 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/109046 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99128 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23160 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126142 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115458 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43830 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32942 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97471 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44195 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101101 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97274 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24767 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42604 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20545 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/40225 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43715 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49311 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144157 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43182 "Built successfully") | | [💥 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37123 "An unexpected error occured. Recent messages:Printed configuration; Running apply-patch; Checked out pull request") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46521 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44887 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.h:
```diff
@@ -26,8 +26,8 @@
 
-#import "AXTextMarker.h"
-#import "WebAccessibilityObjectWrapperBase.h"
+#import <WebCore/AXTextMarker.h>
+#import <WebCore/WebAccessibilityObjectWrapperBase.h>
```
Even within the same project? What made these includes different from the other untouched `#include "..."` cases though?

rr-codes replied:
All private headers must refer to other private headers using angle includes. In project headers and implementation files, it doesn't matter which type you use (though it is better practice in general to use angle brackets for private/public headers)

aprotyas replied:
oh!

aprotyas commented on Source/WebCore/css/typedom/ComputedStylePropertyMapReadOnly.h:
```diff
@@ -25,7 +25,10 @@
 
 #pragma once
 
+#include "Element.h"
```
Is this needed? It's a heavy include

rr-codes replied:
yes; it was already transitively included since the header needs access to the type for 

```
    Element* elementConcurrently() const { return m_element.get(); }
```

aprotyas replied:
Heavy (build time) price to pay for an inline... but makes sense

aprotyas reviewed (commented):
A couple of comments

rr-codes committed:
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 2)
https://bugs.webkit.org/show_bug.cgi?id=297503
rdar://158487517

Reviewed by Abrar Rahman Protyasha.

- Fix some more quoted imports to angle bracket imports
- Convert some more private headers to project that don't need to be private, and vice versa
- Add some missing headers to the Xcode project
- Add a few missing imports
- Let Xcode fix some erroneous file character encoding (macOS Western -> UTF-8)

* Source/WebCore/PAL/pal/text/TextEncoding.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AXTextMarker.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h:
* Source/WebCore/animation/AnimationFrameRatePreset.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/editing/cocoa/AlternativeTextContextController.h:
* Source/WebCore/editing/cocoa/AlternativeTextUIController.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/layout/floats/PlacedFloats.h:
* Source/WebCore/layout/layouttree/LayoutBoxGeometry.h:
* Source/WebCore/page/PerformanceEventTiming.h:
* Source/WebCore/page/ProcessSyncData.in:
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/platform/AudioEncoder.h:
* Source/WebCore/platform/graphics/AV1Utilities.h:
* Source/WebCore/platform/graphics/MediaPlayerEnums.h:

Canonical link: https://commits.webkit.org/298877@main
